### PR TITLE
Deprecate several utility functions

### PR DESCRIPTION
**Pull request by kurtmckee**
This PR introduces the following changes:

* Deprecate two utility functions and their aliases
* Add type annotations to utility functions
* Update utility function doc strings
* Use newer Python constructs to simplify remaining code

### CODE REVIEW

File: src/smartcard/util/__init__.py [hunk 1]
```diff
@@ -22,13 +22,15 @@
...
+from __future__ import annotations
+
 PACK = 1
 HEX = 2
 UPPERCASE = 4
 COMMA = 8
 
 
-def padd(bytelist, length, padding='FF'):
+def padd(bytelist: list[int], length: int, padding: str = 'FF'):
```
Comment (LudovicRousseau) [inline src/smartcard/util/__init__.py:33]: PEP8 https://peps.python.org/pep-0008/#other-recommendations: Don’t use spaces around the = sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter:

Use `str='FF'` without spaces.

File: src/smartcard/util/__init__.py [hunk 2]
```diff
@@ -210,61 +203,21 @@ def toHexString(data=[], format=0):
...
-        if HEX & format:
-            if UPPERCASE & format:
-                pformat = "0X" + pformat
-            else:
-                pformat = "0x" + pformat
-        return (separator.join(map(lambda a: pformat % ((a + 256) % 256), data))).rstrip()
-
-
-# FIXME This appears to duplicate toASCIIString()
-def HexListToBinString(hexlist):
```
Comment (LudovicRousseau) [inline src/smartcard/util/__init__.py:238]: Do NOT remove functions that are part of the documented API
Maybe we should deprecate them first.
I found https://pypi.org/project/Deprecated/ but it is not in the Python 3 standard lib.

**Review by LudovicRousseau — CHANGES_REQUESTED** [review]

File: src/smartcard/util/__init__.py (hunk 1 above)
Comment (kurtmckee) [reply src/smartcard/util/__init__.py:33]: That is true for an unannotated function.

From that same document:

> When combining an argument annotation with a default value, however, do use spaces around the `=` sign:
>
> ```python
> # Correct:
> def munge(sep: AnyStr = None): ...
> def munge(input: AnyStr, sep: AnyStr = None, limit=1000): ...
>
> # Wrong:
> def munge(input: AnyStr=None): ...
> def munge(input: AnyStr, limit = 1000): ...
> ```

File: src/smartcard/util/__init__.py (hunk 2 above)
Comment (kurtmckee) [reply src/smartcard/util/__init__.py:238]: I've updated the PR to follow this. Python support `DeprecationWarning` for this purpose.

**Comment by LudovicRousseau** [comment]
You should also update the documentation to replace calls to deprecated functions

https://github.com/LudovicRousseau/pyscard/blob/master/src/smartcard/doc/user-guide.rst#binary-strings-and-list-of-bytes

**Comment by kurtmckee** [comment]
Shoot, I had been filtering my search to `.py` files. Yep, I'll get that updated. Thanks!

**Comment by kurtmckee** [comment]
Hmm, that section is part of a discussion about cryptography, but specifically pycrypto, which hasn't had a release in 11 years.

I'll make a quick update to that section, but this has informed me that there's some additional work to do with that section on cryptography. :+1:

**Comment by LudovicRousseau** [comment]
You can update the documentation in a different commit.
It will be easier to review.

**Review by LudovicRousseau — APPROVED** [review]

**Comment by kurtmckee** [comment]
I included an update for SHA-1 and MD5 hashing that makes the docs for that section usable.

**Comment by LudovicRousseau** [comment]
Thanks

**Pull request merged**